core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -48,9 +48,21 @@ public abstract class BaseVFSConnectionProvider<T extends VFSConnectionDetails>
   }
 
   @Override public T prepare( T connectionDetails ) throws KettleException {
+    if ( !connectionDetails.getBARoles().contains( "Administrator" ) ) {
```

Comment:
Would be nice to have a doclet saying what this implementation does: ensure that roles contain the `Administrator` role.